libraries/AP_ADSB/AP_ADSB.cpp

```diff
@@ -877,3 +887,27 @@ bool AP_ADSB::get_vehicle_by_ICAO(const uint32_t icao, adsb_vehicle_t &vehicle)
     return false;
 }
 
+/*
+ * Write dataflash log of a vehicle
+ */
+void AP_ADSB::write_log(const adsb_vehicle_t &vehicle)
+{
+    // logging disabled
+    if (_log == 0 || (_log == 1 && !is_special_vehicle(vehicle.info.ICAO_address))) {
```

Comment:
This should probably end up being a switch with an enum. As it stands now, log =-1 and 5 both log.